lib/ProductOpener/Import.pm

```diff
@@ -412,6 +412,167 @@ my %no = (
 );
 
 
+sub apply_organization_settings_to_product($$$$) {
+
+	my %stats = shift;
+	my $org_id = shift;
+	my $org_ref = shift;
+	my $args_ref = shift;
+
+	# Make sure the source as the authorization to load data to the org
+	# e.g. if an org has loaded fresh data manually or through Equadis,
+	# don't overwrite it with potentially stale CodeOnline or USDA data
+
+	if (defined $args_ref->{source_id}) {
+		if (not $org_ref->{"import_source_" . $args_ref->{source_id}}) {
+			$log->debug("skipping import for org without authorization for the source", { org_ref => $org_ref, source_id => $args_ref->{source_id} }) if $log->is_debug();
+			$stats{orgs_without_source_authorization}{$org_id} or $stats{orgs_without_source_authorization}{$org_id} = 0;
+			$stats{orgs_without_source_authorization}{$org_id}++;
+			next;
```

Comment:
this "next" statement was used to get out of the loop that goes through each product, but as it is now in a separate function, it's not going to work.

instead the function should return a value to indicate if the source is authorized to update products from the organization, and if not, we would then call next.

e.g. something like 

if (not source_is_authorized_to_update_products_from_the_org(..) ) {
  next;
}